pyccel/ast/builtins.py

```diff
@@ -655,34 +655,8 @@ class PythonLen(PyccelInternalFunction):
         The argument whose length is being examined.
     """
     __slots__ = ()
-    name      = 'len'
-    _dtype     = NativeInteger()
-    _precision = -1
-    _rank      = 0
-    _shape     = None
-    _order     = None
-    _class_type = NativeInteger()
-
     def __new__(cls, arg):
-        if not getattr(arg, 'is_homogeneous', False):
-            return arg.shape[0]
-        else:
-            return super().__new__(cls)
-
-    def __init__(self, arg):
-        super().__init__(arg)
-
-    @property
-    def arg(self):
-        """
-        Get the argument which was passed to the function.
-
-        Get the argument which was passed to the function.
-        """
-        return self._args[0]
-
-    def __str__(self):
-        return f'len({self.arg})'
```

Comment:
These changes are showing up because you have updated `Set_Support` but haven't merged/rebased the changes into this branch. You will need to pass the changes along through the branches with open PRs